apps/chatbot/src/modules/chatbot.py

```diff
@@ -0,0 +1,204 @@
+from llama_index.embeddings.bedrock import BedrockEmbedding
+
+from src.modules.async_bedrock import AsyncBedrock
+from src.modules.vector_database import load_automerging_index
+from src.modules.retriever import get_automerging_query_engine
+
+
+LANGUAGES = {
+    "it": "Italian",
+    "en": "English",
+    "de": "German",
+    "es": "Spanish",
+    "fr": "French",
+    "pt": "Portuguese",
+    "hr": "Croatian",
+    "sr": "Serbian",
+    "sv": "Swedish",
+    "da": "Danish",
+    "no": "Norwegian",	
+    "fi": "Finnish",
+    "ru": "Russian",
+    "uk": "Ukrainian"
+}
+
+
+QA_PROMPT_STR = """
+You are an Italian customer services chatbot.
+Given the context:
+---------------------
+{context_str}
+---------------------
+Query Rules:
+1. Block any query that requests or involves information that could identify a human being. 
+2. You must give information about yourself, your tasks, etc., when a user asks.
+3. Block queries that are not in Italian.
+4. Block queries containing offensive language, hate speech, or discriminatory content.
+5. Block queries that request speculative or unverified information.
+---------------------
+Answer Rules:
+1. If any of the rules listed above are broken, respond with: "Non posso rispondere a questa domanda perché," followed by a comma-separated list of the violated query rules.
+2. If the retrieved context is empty, reply with: "Mi dispiace, ma non posso fornire una risposta a questa domanda. Per piacere, riformula la domanda in maniera più dettagliata o chiedimene una nuova."
+3. If the query complies with the query rules listed above, generate an answer using only the retrieved context and not any prior knowledge.
+4. The answer must be concise and made by three sentences maximum.
+5. The answers should have no repeations or redundance content.
+6. The answer must be respectful and does not disclose any sensitive or personal information.
+7. The answer must always have at the end a source reference link.
+8. The answer must be in Italian.
+---------------------
+Task:
+Given the user's query: {query_str}
+Generate an answer according to the query and answer rules listed above.
+Answer:
+"""
+
+REFINE_PROMPT_STR = """
+Query: {query_str}
+Existing Answer: {existing_answer}
+---------------------
+Answer Refinement Actions:
+1. The refined answer must better address the question while ensuring compliance with the query rules.
+2. The refined answer must use retrieved context only and do not introduce new information.
+3. The refined answer must not disclose personal or sensitive information.
+4. The refined answer must be respectful and accurate.
+5. The refined answer must not include any offensive or discriminatory content from the existing answer.
+6. The refined answer must not include any repetitions and avoid redundancy.
+7. The refined answer must have a source reference.
+8. The refined answer must be concise and made by maximum three sentences.
+9. The refined answer must must be in Italian.
```

Comment:
```suggestion
9. The refined answer must be in Italian.
```
Is it a typo or a way to be more incisive with the imperative tone?